core/metadata_vec.h

```diff
@@ -0,0 +1,631 @@
+                      const std::vector<MDLabel>* desiredLabels = nullptr);
+    void _readRows(std::istream& is, std::vector<MDObject*> & columnValues, bool useCommentAsImage);
+    /** This function will be used to parse the rows data in START format
+     * @param[out] columnValues MDRow with values to fill in
+     * @param pchStart pointer to the position of '_loop' in memory
+     * @param pEnd  pointer to the position of the next '_data' in memory
+     * @param maxRows if this number if greater than 0, only this number of rows will be parsed.
+     */
+    void _readRowsStar(mdBlock &block, std::vector<MDObject*> & columnValues, const std::vector<MDLabel> *desiredLabels);
+    void _readRowFormat(std::istream& is);
+
+    /**
+     * Get a vector of (empty) objects for each active label
+     */
+    std::vector<MDObject> getObjectsForActiveLabels() const;
+
+
+    /** This two variables will be used to read the metadata information (labels and size)
+     * or maybe a few rows only
+     */
+    size_t _maxRows, _parsedLines;
+
+public:
+    /** @name Constructors
+     *  @{
+     */
+
+    /** Empty Constructor.
+     *
+     * The MetaData is created with no data stored on it. You can fill in it programmatically
+     * or by a later reading from a MetaData file or old Xmipp formatted type.
+     * if labels vectors is passed this labels are created on metadata
+     */
+    MetaDataVec();
+    MetaDataVec(const std::vector<MDLabel> &labelsVector);
+
+    /** From File Constructor.
+     *
+     * The MetaData is created and data is read from provided FileName. Optionally, a vector
+     * of labels can be provided to read just those required labels
+     */
+    MetaDataVec(const FileName &fileName, const std::vector<MDLabel> &desiredLabels);
+    MetaDataVec(const FileName &fileName);
+
+    /** Copy constructor
+     *
+     * Created a new metadata by copying all data from an existing MetaData object.
+     */
+    MetaDataVec(const MetaData &md);
+    MetaDataVec(const MetaDataVec &md) = default;
+
+    /** Assignment operator
+     *
+     * Copies MetaData from an existing MetaData object.
+     */
+    MetaDataVec& operator=(const MetaData &md);
+    MetaDataVec& operator=(const MetaDataVec &md) = default;
+
+    /**Clear all data
+     */
+    void clear() override;
+    /** @} */
+
+    /** @name Getters and setters
+     * @{
+     */
+
+    bool nextBlock(mdBuffer &buffer, mdBlock &block);
+
+    /** Export medatada to xml file.
+     *
+     */
+    void writeXML(const FileName fn, const FileName blockname, WriteModeMetaData mode) const override;
+
+    /** Write metadata in text file as plain data without header.
+     *
+     */
+    void writeText(const FileName fn,  const std::vector<MDLabel>* desiredLabels) const override;
+
+    void _parseObjects(std::istream &is, std::vector<MDObject*> & columnValues, const std::vector<MDLabel> *desiredLabels, bool firstTime);
```

Comment:
these shouold be private, protected at best

Reply:
Both? `parseObjects` moved to `protected`.